src/data/list/forall2.lean

```diff
@@ -219,4 +219,77 @@ lemma rel_prod [monoid α] [monoid β]
   (h : r 1 1) (hf : (r ⇒ r ⇒ r) (*) (*)) : (forall₂ r ⇒ r) prod prod :=
 rel_foldl hf h
 
+/-- Given a relation `r`, `sublist_forall₂ r l₁ l₂` indicates that there is a sublist of `l₂` such
+  that `forall₂ r l₁ l₂`. -/
+inductive sublist_forall₂ (r : α → β → Prop) : list α → list β → Prop
+| nil {l} : sublist_forall₂ [] l
+| cons {a₁ a₂ l₁ l₂} : r a₁ a₂ → sublist_forall₂ l₁ l₂ →
+  sublist_forall₂ (a₁ :: l₁) (a₂ :: l₂)
+| cons_right {a l₁ l₂} : sublist_forall₂ l₁ l₂ → sublist_forall₂ l₁ (a :: l₂)
+
+lemma sublist_forall₂_iff {l₁ : list α} {l₂ : list β} :
+  sublist_forall₂ r l₁ l₂ ↔ ∃ l, forall₂ r l₁ l ∧ l <+ l₂ :=
```

Comment:
Is this inductive definition easier to work with than using `∃ l, forall₂ r l₁ l ∧ l <+ l₂` directly?

Reply:
I think it is easier for, say, the transitivity proof, but I realize that I could cut down the proof of reflexivity. The inductive definition also follows Minchao Wu's work: https://github.com/minchaowu/Kruskal.lean3

Reply:
@awainverse There are actually two version’s of Higman’s lemma formalized in that repo. One is using this `sublist` notion, and the other is using sets. Is this `sublist` version easier to work with than the set version (for your purpose)?

Reply:
It is easier to work with for my purposes, matches the sources that I've seen closer, and hopefully my proof generalizes to `finset` or `multiset` versions without too much difficulty.